ldapAdapterEXT-ext-impl/src/main/java/com/bosch/osmi/sw360/ldapAdapter/CustomPortalLDAPImporterImpl.java

```diff
@@ -8,20 +8,47 @@
 import com.liferay.portal.security.ldap.LDAPSettingsUtil;
 import com.liferay.portal.security.ldap.PortalLDAPImporterImpl;
 import com.liferay.portal.security.membershippolicy.OrganizationMembershipPolicyUtil;
-import com.liferay.portal.service.*;
+import com.liferay.portal.service.OrganizationLocalService;
+import com.liferay.portal.service.OrganizationLocalServiceUtil;
+import com.liferay.portal.service.UserLocalServiceUtil;
 import org.apache.log4j.Logger;
 
 import javax.naming.NamingException;
 import javax.naming.directory.Attribute;
 import javax.naming.directory.Attributes;
 import javax.naming.ldap.LdapContext;
-import java.util.Collections;
-import java.util.Properties;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.stream.Collectors;
 import java.util.stream.LongStream;
 
+import static org.apache.log4j.LogManager.getLogger;
+
 public class CustomPortalLDAPImporterImpl extends PortalLDAPImporterImpl {
+	private static final String MAPPING_KEYS_PREFIX = "mapping.";
+	private static final String MAPPING_VALUES_SUFFIX = ".target";
+	private boolean matchPrefix = false;
+	private Map<String, String> organizationMappings;
```

Comment:
To get an additional level of fancyness one could save this data structure as search-tree.